core/jvm/src/main/java/zio/internal/Signal.java

```diff
@@ -0,0 +1,130 @@
+package zio.internal;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Proxy;
+import java.util.function.Consumer;
+
+
+/**
+ * This class is used to handle system signals in a platform-specific way. It attemps to use the
+ * sun.misc.Signal and sun.misc.SignalHandler classes if they are available. If they are not
+ * available, it will gracefully degrade to a no-op.
+ * <p>
+ * <b>IMPLEMENTATION NOTE</b>: This class uses reflection to access the sun.misc.Signal and sun.misc.SignalHandler.
+ * This is necessary because the sun.misc.Signal and sun.misc.SignalHandler classes might not be available in some
+ * JVMs.
+ * </p>
+ */
+final class Signal {
+
+    private Signal() {
+    }
+
+    private static final MethodHandle SIGNAL_CONSTRUCTOR_METHOD_HANDLE;
+    private static final MethodHandle HANDLE_STATIC_METHOD_HANDLE;
+    private static final SignalHandler SIGNAL_HANDLER;
+
+    static {
+        final Class<?> signalClass = findClass("sun.misc.Signal");
+        final Class<?> signalHandlerClass = findClass("sun.misc.SignalHandler");
+
+        if ((signalClass != null) && (signalHandlerClass != null)) {
+            final MethodHandles.Lookup lookup = MethodHandles.lookup();
+
+            SIGNAL_CONSTRUCTOR_METHOD_HANDLE = initSignalConstructorMethodHandle(lookup, signalClass);
+            HANDLE_STATIC_METHOD_HANDLE = initHandleStaticMethodHandle(lookup, signalClass, signalHandlerClass);
+            SIGNAL_HANDLER = new SignalHandler.SunMiscSignalHandler(signalHandlerClass);
+        } else {
+            System.err.println("WARNING: sun.misc.Signal and sun.misc.SignalHandler are not available on this platform. " +
```

Comment:
Couple of questions here:

1. Should we use a logger instead of `System.err.println`?
2. Should we add _some_ option to disable this warning? Might be annoying in some cases, but I think as a default a user should be aware that the fiber dump functionality won't work as expected

Reply:
Thanks for the recommendation! I switched to using `java.logging` to log the warning and changed the code to make it a bit more readable / safe so that we log the warning if any of the reflective methods fail